src/pybind/mgr/dashboard/frontend/src/app/shared/components/empty-state/empty-state.component.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+
+import { Icons } from '~/app/shared/enum/icons.enum';
+
+@Component({
+  selector: 'cd-empty-state',
```

Comment:
> @epuertat The problem I would face with that is there is currently no way I could make the button perform the necessary action I need to configure the rbd-mirroring (in this case its creating the service and creating the pool). because the module-status-guard will redirect to error page based on the status and from the error page its very limited option. (currently its just route to different page and we can customise the routeURL). But I can't make it perform any complex actions using the error.component. So I won't be able to use the module-status-guard to redirect.
> 
> What I can use is maybe customise the error.component so that I can call it with a `cd-error-page` thereby it'll be a unique layout for all empty-states. I haven't tested it though.

@nizamial09 Got it. Checking the code from @melissa-kun-li here: https://github.com/ceph/ceph/pull/46407/files#diff-0ce5435c0908787b826bc5f3d957e40fe8f91bbab1c303a2b1ccb18d5c98ca00L29-L30, she's providing this new `button_route`, which is a UI route. This leaves us with these options:
* **Front-end approach 1**: provide a UI route to a multi-stage wizard... This would involve creating a generic Wizard component made of multiple forms (rbdmirror service creation, rbd pool creation). However, the approach we were suggesting here was to pre-create the services without user interaction, so we don't really need such a wizard or forms.
* **Front-end approach 2**: provide a UI route that performs what your doing here in the background. Maybe we could use a query string parameter (`?bootstrap=1`) to tell the RBD Mirroring page to trigger the service creation during ngOnInit (the challenge here would be that the service status guard would be run first and block it, so we would need to figure out how to skip the guard under some circumstances).
* **Front-end approach 3**: instead of a URL, we move the RBD mirroring URL-query param to a service, and from the error component-empty state button click callback we call that service (I'm not sure if services/methods can be passed as arguments for guards). 
* **Back-end approach**: we could extend Melissa's approach with an extra approach that instead of routing the UI to another component, directly performs a POST/PUT action to a back-end endpoint (and that endpoint fully takes care of doing the service and pool creation).

The benefit of this is leveraging patterns instead of applying ad-hoc recipes. We should always challenge the first impression of "this can't be done" or "this case is an exception":
* Sometimes things can be done with an existing pattern if you simply shift a bit your initial approach.
* Sometimes things definitely can't be done with a pattern, but exceptions are hard to maintain in the long term. The benefit of patterns is that they are easy to extend to every other component, and if you ensure that the pattern is properly tested you don't need to create tests for every occurrence of that pattern (unlike what we should do with ad-hoc customizations).